src/test/scala/kinesis/mock/cache/PutRecordTests.scala

```diff
@@ -68,12 +69,9 @@ class PutRecordTests
             .getRecords(GetRecordsRequest(None, shardIterator), context, false)
             .rethrow
         } yield assert(
-          res.records.length == 5 && res.records.forall(rec =>
-            recordRequests.exists(req =>
-              req.data.sameElements(rec.data)
-                && req.partitionKey == rec.partitionKey
-            )
-          ),
+          res.records.length == 5 && res.records.toVector.map(
+            PutRecordResults.fromKinesisRecord
+          ) === recordRequests.map(PutRecordResults.fromPutRecordRequest),
```

Comment:
Changed this test to be more strict, and ensure that the ordering of the records are the same.